src/context/SeasonContext.tsx

```diff
@@ -0,0 +1,67 @@
+import {
+  FC,
+  ReactElement,
+  createContext,
+  useCallback,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+
+export const Seasons = {
+  Winter: 'Winter',
+  Spring: 'Spring',
+  Summer: 'Summer',
+  Fall: 'Fall',
+} as const; // use const instead of enum to use Object.keys
+
+export type Season = keyof typeof Seasons;
+type SeasonContextType = {
+  season: Season;
+  nextSeason: () => void;
+};
+
+export const AllSeasons = Object.keys(Seasons) as Season[];
+
+const SeasonContext = createContext<SeasonContextType>({
+  season: Seasons.Spring,
+  nextSeason: () => {
+    throw new Error(
+      'The SeasonContext has been used before its initialization',
+    );
+  },
+});
+
+type Prop = {
+  children: ReactElement;
+};
+
+export const SeasonProvider: FC<Prop> = ({ children }) => {
```

Comment:
I would suggest to not use FC.
```suggestion
export const SeasonProvider = ({ children }: Prop) => {
```